# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/COLRFonts.h
# Commit: 4354a0d7248f
# Full Hash: 4354a0d7248f50dde9ba6bcf6149eda57b7c9bb3
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2022-08-17 15:40:28
# Regressor Bug: 1740530
# File Overlap Count: 1
# Description:
#   Bug 1740530 - patch 4 - Use harfbuzz color-palette access functions, instead of rolling our own. r=gfx-reviewers,lsalzman
#   
#   Depends on D152038
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D153870
# ==============================================================================

diff -r 5d1de3ed837b -r 4354a0d7248f gfx/thebes/COLRFonts.h
--- a/gfx/thebes/COLRFonts.h	Wed Aug 17 10:59:34 2022 +0000
+++ b/gfx/thebes/COLRFonts.h	Wed Aug 17 10:59:34 2022 +0000
@@ -9,6 +9,7 @@
 #include "mozilla/gfx/2D.h"
 
 struct hb_blob_t;
+struct hb_face_t;
 
 namespace mozilla {
 
@@ -28,7 +29,7 @@
                                                    uint32_t aGlyphId);
 
   static bool PaintGlyphLayers(
-      hb_blob_t* aCOLR, hb_blob_t* aCPAL, const COLRBaseGlyphRecord* aLayers,
+      hb_blob_t* aCOLR, hb_face_t* aFace, const COLRBaseGlyphRecord* aLayers,
       DrawTarget* aDrawTarget, layout::TextDrawTarget* aTextDrawer,
       ScaledFont* aScaledFont, DrawOptions aDrawOptions,
       const sRGBColor& aCurrentColor, const Point& aPoint);